### PR TITLE
Reversed guessing game.

### PR DESCRIPTION
**Pull request by Priyanka15-12**
Reversed method of guessing game.

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -1,7 +1,32 @@
...
+   private static void Main (string[] args) {
+      int a = 1, z = 100, guess;
+      Console.Write ($"Guess random number from {a} to {z} and press enter key.");
+      Console.ReadKey ();
+      int attempt = 1;
+      while (true) {
+         guess = (a + z) / 2;
+         char key;
+         Console.Write ($"\nIs {guess} your answer?(Y/N): ");
+         key = Console.ReadKey ().KeyChar;
```
Comment (MetaNaveen) [inline Program.cs:13]: For simple applications, like this, it's good to consider case-insensitive answer from users (i.e.,. 'n' | 'N'). Change it accordingly.

File: Program.cs [hunk 2]
```diff
@@ -1,7 +1,32 @@
...
+         if (key == 'Y') {
+            Console.Write ($"\n{guess} is your guessed random number and I found it in {attempt} attempt.");
```
Comment (MetaNaveen) [inline Program.cs:15]: Can you use [Ternary Operator](https://www.w3schools.com/cs/cs_conditions_shorthand.php) to print either "attempt" or "attempts" to make it grammatically correct?

File: Program.cs [hunk 3]
```diff
@@ -1,7 +1,31 @@
...
+using System;
+internal class Program {
+   private static void Main (string[] args) {
+      int a = 1, z = 100, guess;
+      Console.Write ($"Guess random number from {a} to {z} and press enter key.");
+      Console.ReadKey ();
+      int attempt = 1;
+      while (true) {
+         guess = (a + z) / 2;
+         char key;
```
[the remark this answers is not in the document]
Comment (MetaNaveen) [reply Program.cs:9]: char key = Console.ReadKey().KeyChar; could have been used.

File: Program.cs [hunk 4]
```diff
@@ -1,7 +1,29 @@
...
-            Console.WriteLine ("Hello, World!");
-        }
-    }
+using System;
+internal class Program {
+   private static void Main (string[] args) {
+      int a = 1, z = 100, attempt = 1, guess;
+      Console.Write ($"Guess random number from {a} to {z+1} and press enter key.");
+      Console.ReadKey ();
+      while (true) {
```
Comment (MetaNaveen) [inline Program.cs:7]: If user uses No as answer, the output keep on repeating "Is that lesser than 100? Y/N: AND
Is 100 your answer?(Y/N):". 
Can you think and set maximum number of iteration before telling user "Sorry I failed to find your out-of-the-box guessing!"

**Review by MetaNaveen — CHANGES_REQUESTED** [review]

File: Program.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (Priyanka15-12) [reply Program.cs]: I changed this.

File: Program.cs (hunk 4 above)
Comment (Priyanka15-12) [reply Program.cs]: Repetition of questions reduced.

**Comment by MetaNaveen** [comment]
Working fine.

**Points to consider:**
Since maximumGuess is hardcoded and used in multiple places. You could store it in a variable and use it.
Maximum guess can be computed using the formula **Math.Ceiling(Math.Log2(max - min + 1))**

**Pull request closed without merging**